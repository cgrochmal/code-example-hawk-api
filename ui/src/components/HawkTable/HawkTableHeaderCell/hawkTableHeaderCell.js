import React from 'react'
import PropTypes from 'prop-types'

import './hawkTableHeaderCell.scss'

HawkTableHeaderCell.propTypes = {
  // currently supported cellTypes are 'name', 'size', and 'gender'
  cellType: PropTypes.string.isRequired,
  // called with (cellType, 'asc'/'desc')
  handleSortChange: PropTypes.func
}

/**
 * Helper component representing an individual header cell in HawkTable
 */
export default function HawkTableHeaderCell(props) {
  const {handleSortChange, cellType} = props
  return (
    <span className={'header-cell header-cell--'+cellType}>
      <div className='header-cell__name'>
        {cellType}
      </div>
      <div className='header-cell__arrows'>
        <div className='arrow up-arrow' onClick={() => handleSortChange(cellType, 'asc')}>&#x25B2;</div>
        <div className='arrow down-arrow' onClick={() => handleSortChange(cellType, 'desc')}>&#x25BC;</div>
      </div>
    </span>
  )
}