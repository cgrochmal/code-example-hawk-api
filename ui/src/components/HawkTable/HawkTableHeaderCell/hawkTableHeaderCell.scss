.header-cell {
  box-sizing: border-box;
  padding: 8px;
  border-right: 1px solid;
  background: #aaa;
  &__name {
    display: inline-block;
  }
  &__arrows {
    float: right;
    .arrow {
      height: 13px;
    }
  }
}