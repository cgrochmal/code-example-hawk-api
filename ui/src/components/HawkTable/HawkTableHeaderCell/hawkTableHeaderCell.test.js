import React from 'react'
import renderer from 'react-test-renderer'
import HawkTableHeaderCell from './hawkTableHeaderCell'

it('matches the HawkTableHeaderCell snapshot', () => {
  const tree = renderer
    .create(<HawkTableHeaderCell cellType={'name'}/>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})