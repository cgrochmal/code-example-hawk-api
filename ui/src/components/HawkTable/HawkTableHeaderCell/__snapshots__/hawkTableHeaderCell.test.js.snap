// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the HawkTableHeaderCell snapshot 1`] = `
<span
  className="header-cell header-cell--name"
>
  <div
    className="header-cell__name"
  >
    name
  </div>
  <div
    className="header-cell__arrows"
  >
    <div
      className="arrow up-arrow"
      onClick={[Function]}
    >
      ▲
    </div>
    <div
      className="arrow down-arrow"
      onClick={[Function]}
    >
      ▼
    </div>
  </div>
</span>
`;
